IdentityCore/src/broker_operation/response/MSIDBrokerOperationGetSsoCookiesResponse.m

```diff
@@ -61,20 +61,34 @@ - (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(NSError **)er
         }
         
         NSString *ssoCookiesString = json[@"sso_cookies"];
-        
         if ([NSString msidIsStringNilOrBlank:ssoCookiesString])
         {
-            self.prtHeaders = nil;
-            self.deviceHeaders = nil;
-            self.success = YES;
             return self;
         }
         
         NSDictionary *ssoCookiesJson = [ssoCookiesString msidJson];
+        NSError *convertPrtHeaderError = nil;
+        NSError *convertDeviceHeaderError = nil;
+        _prtHeaders = [self convertToCredentialHeaderObjectFrom:ssoCookiesJson credentialName:MSID_PRT_HEADERS error:&convertPrtHeaderError];
+        _deviceHeaders = [self convertToCredentialHeaderObjectFrom:ssoCookiesJson credentialName:MSID_DEVICE_HEADERS error:&convertDeviceHeaderError];
         
-        _prtHeaders = [self convertToCredentialHeaderObjectFrom:ssoCookiesJson credentialName:MSID_PRT_HEADERS error:error];
-        
-        _deviceHeaders = [self convertToCredentialHeaderObjectFrom:ssoCookiesJson credentialName:MSID_DEVICE_HEADERS error:error];
+        if(convertPrtHeaderError && convertDeviceHeaderError)
+        {
+            if(error)
+            {
+                *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, [NSString stringWithFormat:@"Cannot convert credential headers from json response. PrtHeader error is %@. Device header error is %@", convertPrtHeaderError.description, convertDeviceHeaderError.description], nil, nil, nil, nil, nil, YES);
+            }
+            
+            self.success = NO;
```

Comment:
What is the purpose of self.success, is it being used somewhere else? I don't see it's used anywhere in this PR, so just curious.

Reply:
Yes, it will be user in this [PR](https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/pull/1056)

In general, we will check if the response is successful or not.

if not successful, return error back to caller
if successful, return result back to caller